spanner/jdbc/src/main/java/com/example/spanner/jdbc/PgCaseSensitivitySample.java

```diff
@@ -0,0 +1,124 @@
+                "jdbc:cloudspanner:/projects/%s/instances/%s/databases/%s",
+                projectId, instanceId, databaseId))) {
+
+      // Spanner PostgreSQL follows the case sensitivity rules of PostgreSQL. This means that:
+      // 1. Identifiers that are not double-quoted are folded to lower case.
+      // 2. Identifiers that are double-quoted retain their case and are case-sensitive.
+      // See https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS
+      // for more information.
+
+      connection
+          .createStatement()
+          .execute(
+              "CREATE TABLE Singers ("
+                  // SingerId will be folded to `singerid`.
+                  + "  SingerId      bigint NOT NULL PRIMARY KEY,"
+                  // FirstName and LastName are double-quoted and will therefore retain their
+                  // mixed case and are case-sensitive. This means that any statement that
+                  // references any of these columns must use double quotes.
+                  + "  \"FirstName\" varchar(1024) NOT NULL,"
+                  + "  \"LastName\"  varchar(1024) NOT NULL"
+                  + ")");
+
+      connection
+          .unwrap(CloudSpannerJdbcConnection.class)
+          .write(
+              Collections.singleton(
+                  Mutation.newInsertBuilder("Singers")
+                      .set("singerid")
+                      .to(1L)
+                      // Column names in mutations are always case-insensitive, regardless whether
+                      // the
+                      // columns were double-quoted or not during creation.
+                      .set("firstname")
+                      .to("Bruce")
+                      .set("lastname")
+                      .to("Allison")
+                      .build()));
+
+      try (ResultSet singers = connection.createStatement().executeQuery("SELECT * FROM Singers")) {
+        while (singers.next()) {
+          System.out.printf(
+              "SingerId: %d, FirstName: %s, LastName: %s\n",
+              // SingerId is automatically folded to lower case. Accessing the column by its name in
+              // a result set must therefore use all lower-case letters.
+              singers.getLong("singerid"),
+              // FirstName and LastName were double-quoted during creation, and retain their mixed
+              // case when returned in a result set.
+              singers.getString("FirstName"),
+              singers.getString("LastName"));
+        }
+      }
+
+      // Aliases are also identifiers, and specifying an alias in double quotes will make the alias
+      // retain its case.
+      try (ResultSet singers =
+          connection
+              .createStatement()
+              .executeQuery(
+                  "SELECT "
+                      + "singerid AS \"SingerId\", "
+                      + "concat(\"FirstName\", ' '::varchar, \"LastName\") AS \"FullName\" "
```

Comment:
is the ::varchar here to workaround the concat issue?

you can also use  ||  which probably doesn't require the varchar casting.

Reply:
> is the ::varchar here to workaround the concat issue?

I'm not entirely sure what "the concat issue" is, but this is a workaround for the fact that the `concat` function takes arguments of type `any`. That means that the PostgreSQL parser will interpret the string literals as having type `unknown`, which is not supported by Spangres.

> you can also use || which probably doesn't require the varchar casting.

Good point. I've changed it to using `||` instead, as this sample is not about function calls and specific casting required for some of those.

Reply:
That is the exactly `concat` issue I mean.  We are working on some enhancement on it.